File: two_sum/two_sum.py
"""Given an array of integers, return *indices* of the two numbers such that they add up to the specific target

You may assume that each input would have exactly one solution, and you may not use the same element twice."""

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index = {}
        for i, num in enumerate(nums):
            other = target - num
            if other in index:
                return [i, index[other]]
            index[num] = i

File: two_sum/two_sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            int other = target - nums[i];
            if(map.containsKey(other)) {
                return new int[] {map.get(other), i};
            }
            map.put(nums[i], i);
        }
        throw new IllegalArgumntException("No two sum solution");
    }
}
